app/1.0.x/plugin-development/admin-api.md

```diff
@@ -34,100 +34,138 @@ Kong will detect and load your endpoints if they are defined in a module named:
 "kong.plugins.<plugin_name>.api"
 ```
 
-This module is bound to return a table containing strings describing your
-routes (See [Lapis routes & URL
-Patterns](http://leafo.net/lapis/reference/actions.html#routes--url-patterns))
-and HTTP verbs they support. Routes are then assigned a simple handler
-function.
+This module is bound to return a table with one or more entries with the following structure:
+
+``` lua
+{
+  ["<path>"] = {
+     schema = <schema>,
+     methods = {
+       before = function(self) ... end,
+       on_error = function(self) ... end,
+       GET = function(self) ... end,
+       PUT = function(self) ... end,
+       ...
+     }
+  },
+  ...
+}
+```
 
-This table is then fed to Lapis (See Lapis' [handling HTTP verbs
-documentation](http://leafo.net/lapis/reference/actions.html#handling-http-verbs)).
-Example:
+Where:
+
+- `<path>` should be a string representing a route like `/users` (See [Lapis routes & URL
+  Patterns](http://leafo.net/lapis/reference/actions.html#routes--url-patterns)) for details.
+  Notice that the path can contain interpolation parameters, like `/users/:users/new`.
+- `<schema>` is a schema definition. Schemas for core and custom plugin entities are available
+  via `kong.db.<entity>.schema`. The schema is used to parse certain fields according to their
+  types; for example if a field is marked as an integer, it will be parsed as such when it is
+  passed to a function (by default form fields are all strings).
+- The `methods` subtable contains functions, indexed by a string.
+  - Zero or one function can be indexed with `before`. If present, the function will be executed
```

Comment:
"Zero or one function" sounds very computery. I appreciate the formal-grammar-like precision, but given that Lua tables can only have one entry with a given name anyway, it would read better if it just said `The "before" key is optional and can hold a function. If present, it will be executed...`